learnify/mobile/lib/features/auth/forget-password/view-model/forget_password_view_model.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:async/async.dart';
+import 'package:flutter/material.dart';
+
+import '../../../../core/base/view-model/base_view_model.dart';
+
+class ForgetPasswordViewModel extends BaseViewModel {
+  late TextEditingController _emailController;
+
+  TextEditingController get emailController => _emailController;
+
+  late GlobalKey<FormState> _formKey;
+  GlobalKey<FormState> get formKey => _formKey;
+
+  bool _canVerify = false;
+  bool get canVerify => _canVerify;
+
+  @override
+  void initViewModel() {}
+
+  @override
+  void initView() {
+    _formKey = GlobalKey<FormState>();
+    _emailController = TextEditingController();
+    _emailController.addListener(_controllerListener);
+    _setDefault();
+  }
+
+  @override
+  void disposeView() {
+    _emailController.dispose();
+    _setDefault();
+    super.disposeView();
+  }
+
+  void _controllerListener() {
+    final bool newCanVerify = _emailController.text.isNotEmpty;
+    if (_canVerify == newCanVerify) return;
+    _canVerify = newCanVerify;
+    notifyListeners();
+  }
+
+  Future<String?> forgetPassword() async {
+    await operation?.cancel();
+    operation =
+        CancelableOperation<String?>.fromFuture(_forgetPasswordRequest());
+    final String? res = await operation?.valueOrCancellation();
+    return res;
+  }
+
+  Future<String?> _forgetPasswordRequest() async {
+    final bool isValid = formKey.currentState?.validate() ?? false;
+    if (isValid) {
+      return null;
```

Comment:
You can remove this line since it is unnecessary.

Reply:
I mean when you just write `return null` at the end of the function, it will have the same behaviour with the one you did. Both implementation will return null but your one has 1 unnecessary line inside if block. I meant to remove "return null" statement inside if block.

Reply:
Since this part is already being modified with the backend connection anyway I guess it doesn't really matter. Still, I can remove it here as well just so it looks cleaner